chain/chain/src/chain.rs

```diff
@@ -4257,9 +4257,10 @@ impl<'a> ChainUpdate<'a> {
         let prev_header = self.get_previous_header(header)?.clone();
 
         // Check that epoch_id in the header does match epoch given previous header (only if previous header is present).
-        if &self.runtime_adapter.get_epoch_id_from_prev_block(header.prev_hash())?
-            != header.epoch_id()
-        {
+        let epoch_id_from_prev_block =
```

Comment:
this doesn't seem to be related to the PR title. (or is it just a drive-by refactor? ;-)

Reply:
yes, it's a drive by refactor. It was easier to debug in this version